tools/shared.py

```diff
@@ -1901,6 +1901,7 @@ def link_lld(args, target, opts=[], lto_level=0):
         '--export',
         '__data_end',
         '--lto-O%d' % lto_level,
+        '--mllvm', '-disable-lsr', # see binaryen#1054
```

Comment:
Rather than adding it here I think you want to add it to Building.llvm_backend_args() (which gets applied below) since that will effect both LTO and separate compilation.